src/Engines/Nunjucks.js

```diff
@@ -4,6 +4,78 @@ const TemplatePath = require("../TemplatePath");
 const EleventyErrorUtil = require("../EleventyErrorUtil");
 const EleventyBaseError = require("../EleventyBaseError");
 
+/*
+ * HACKITYHACKHACKHACKHACK
+ */
+let inc = ((i) => {
+  return () => {
+    return i++;
+  };
+})(0);
+let pathMap = new Map();
+
+let clearCache = (reason = "") => {
+  if (pathMap.size > 0) {
+    pathMap.clear();
+  }
+};
+
+let clearAndWrap = (name, obj = NunjucksLib.Environment) => {
+  let orig = obj.prototype[name];
+  obj.prototype[name] = function (...args) {
+    clearCache(name);
+    return orig.call(this, ...args);
+  };
+};
+
+clearAndWrap("addExtension");
+clearAndWrap("removeExtension");
+
+(function () {
+  // IFFE to prevent temp var leakage
+  let getKey = (obj) => {
+    let eids = obj.env.extensionsList.reduce((v, c) => {
+      return c.__id ? v + " : " + c.__id : v;
```

Comment:
I'd use a template literal for the positive case.

Reply:
`c` is an added extension implementation?
What does `v` stands for? (It's an assembled string …)

Reply:
Will rename.

Reply:
```suggestion
      return c.__id ? `${v} : ${c.__id}` : v;
```

Reply:
reworked in updated patch to avoid all of this.